src/server/nfc/index.ts

```diff
@@ -0,0 +1,159 @@
+import { UserInputError, AuthenticationError } from 'apollo-server-express';
+import { ObjectID, FindAndModifyWriteOpResultObject } from 'mongodb';
+import {
+	HackerDbObject,
+	OrganizerDbObject,
+	EventDbObject,
+	EventCheckInDbObject,
+	ShirtSize,
+	UserDbInterface,
+	UserInput,
+	UserType,
+} from '../generated/graphql';
+import Context from '../context';
+import {
+	fetchUser,
+	query,
+	queryById,
+	toEnum,
+	updateUser,
+	checkIsAuthorized,
+} from '../resolvers/helpers';
+import { Models } from '../models';
+
+// TODO: (kenli/timliang) Expand functions for Organizers, Mentors collections
+
+export async function checkIfNFCUIDExisted(
+	nfcID: string,
+	models: Models
+): Promise<HackerDbObject | null> {
+	const hacker = await models.Hackers.findOne({
+		secondaryIds: nfcID,
+	});
+	return hacker;
+}
+
+export async function getUser(nfcID: string, models: Models): Promise<HackerDbObject | null> {
+	const user = await models.Hackers.findOne({
+		'secondaryIds.0': nfcID,
+	});
+	return user;
+}
+
+export async function isNFCUIDAvailable(nfcID: string, models: Models): Promise<boolean> {
+	return (await getUser(nfcID, models)) == null;
+}
+
+export async function registerNFCUIDWithUser(
+	nfcID: string,
+	userID: string,
+	models: Models
+): Promise<string | null> {
+	if (await isNFCUIDAvailable(nfcID, models)) {
+		const ret = await models.Hackers.findOneAndUpdate(
+			{ _id: new ObjectID(userID) },
+			{
+				$push: {
+					secondaryIds: {
+						$each: [nfcID],
+						$position: 0,
+					},
+				},
+			}
+		);
+		if (ret.value !== undefined) return userID;
+	}
+	return null;
+}
+
+export async function removeUserFromEvent(
+	userID: string,
+	eventID: string,
+	models: Models
+): Promise<string | null> {
+	const user = await models.Hackers.findOne({ _id: new ObjectID(userID) });
+	if (user) {
+		const nfcUID = user.secondaryIds[0];
+		const event = await models.Events.findOne({ attendees: nfcUID });
```

Comment:
When you are removing users from events' attendees you look for nfcUID but when you are adding them to attendees you are adding by userID. Let's store attendees by userID.

Reply:
Attendees stored by userID : ObjectID. eventsAttended also stores eventID : ObjectID